modules/light-clients/06-solomachine/client_state.go

```diff
@@ -118,55 +118,16 @@ func (cs *ClientState) VerifyMembership(
 	path []byte,
 	value []byte,
 ) error {
-	// TODO: Attempt to refactor code to smaller function
-	if revision := height.GetRevisionNumber(); revision != 0 {
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidHeight, "revision must be 0 for solomachine, got revision-number: %d", revision)
-	}
-
-	// sequence is encoded in the revision height of height struct
-	sequence := height.GetRevisionHeight()
-	latestSequence := cs.GetLatestHeight().GetRevisionHeight()
-	if latestSequence != sequence {
-		return sdkerrors.Wrapf(
-			sdkerrors.ErrInvalidHeight,
-			"client state sequence != proof sequence (%d != %d)", latestSequence, sequence,
-		)
+	publicKey, sigData, timestamp, sequence, err := produceVerificationArgs(cdc, cs, height, proof)
```

Comment:
Sounds like a great suggestion to me

I guess my only concern would be constructing a type `verificationArgs` which are dependent on arguments that needed to be checked rather than the totality of arguments required for verification. The path/diversifier/data being the others. We could also take away some of the return arguments (public key, sequence) can be derived from the client state. Originally this function was made to reduce redundancy of calls by 6-7 functions, now only 2 functions rely on it so redundancy isn't as much of a concern. Thoughts?

Reply:
In general I agree, and normally lean against having such a large number of return args. I'd be happy to continue the discussion or explore options for improving the function sig and its usage, maybe we could potentially open an issue for this? At least this an unexported function so whatever we decide should not have any implications on consumers. 
I'm going to go ahead and merge this to the [WIP] branch and get it R4R with renaming types to remove the `V2` suffixes.